### PR TITLE
feat: project page overview V2 (data sources)

### PR DESCRIPTION
**Pull request by andre-code**
PR to add ata sources in a v2 project.
Should be able to add, remove, see details, edit a data source.

data source

/deploy renku=release-0.52.x

### CODE REVIEW

File: client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+export function DataSourceDisplay({
+  storage,
+  projectId,
+}: DataSourceDisplayProps) {
+  const storageSensitive = storage.storage;
+  const [toggleView, setToggleView] = useState(false);
+  const toggleDetails = () => {
+    setToggleView((open: boolean) => !open);
+  };
```
Comment (leafty) [inline client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx:204]:
```suggestion
  const toggleDetails = useCallback(() => {
    setToggleView((open: boolean) => !open);
  }, []);
```

File: client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  const nameDisplay = (
+    <div
+      className={cx(
+        "d-flex",
+        "align-items-center",
+        "gap-1",
+        "cursor-pointer",
+        sessionItemStyles.ItemDisplaySessionName
+      )}
+      onClick={() => toggleDetails()}
```
Comment (leafty) [inline client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx:215]:
```suggestion
      onClick={toggleDetails}
```

File: client/src/features/projectsV2/api/projectV2.enhanced-api.ts [hunk 3]
```diff
@@ -114,6 +131,7 @@ const enhancedApi = injectedApi.enhanceEndpoints({
     "Namespace",
     "Project",
     "ProjectMembers",
+    "Storages",
```
Comment (leafty) [inline client/src/features/projectsV2/api/projectV2.enhanced-api.ts:134]:
```suggestion
    "Storage",
```

File: client/src/features/projectsV2/api/storages.api-config.ts
Comment (leafty) [inline client/src/features/projectsV2/api/storages.api-config.ts:1]: `storage` API. "Storages" is not an English word.

File: client/.eslintrc.json [hunk 4]
```diff
@@ -238,6 +240,7 @@
           "statuspage",
           "stdout",
           "stockimages",
+          "storages",
```
Comment (leafty) [inline client/.eslintrc.json:243]: "storages" is not proper English.

```suggestion
```

File: client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+          <DropdownItem
+            className={cx(
+              "d-flex",
+              "align-items-center",
+              "gap-2",
+              "justify-content-start"
+            )}
+            onClick={toggleEdit}
+          >
+            <Pencil /> Edit data sources
```
Comment (leafty) [inline client/src/features/ProjectPageV2/ProjectPageContent/DataSources/DataSourceDisplay.tsx:172]:
```suggestion
            <Pencil /> Edit data source
```

**Review by leafty — CHANGES_REQUESTED** [review]
Code looks OK. :raised_hands: 

Can you do a contrast check here. The labels do not seem to have enough contrast here.
Screenshot from 2024-05-08 15-08-45

Comment: we have so much space here, why do we not have the actions directly available as buttons? It feels annoying to do two clicks.
Screenshot from 2024-05-08 15-08-57

File: client/src/features/projectsV2/api/storages.api-config.ts
Comment (andre-code) [reply client/src/features/projectsV2/api/storages.api-config.ts:1]: Even though 'storages' isn't used much, it's still proper English. In our data services API, we refer to the storage as 'storages'. So, for now, I'll stick with that name to keep things consistent

**Comment by andre-code** [comment]
> Code looks OK. 🙌
> 
> Can you do a contrast check here. The labels do not seem to have enough contrast here. Screenshot from 2024-05-08 15-08-45
> 
> Comment: we have so much space here, why do we not have the actions directly available as buttons? It feels annoying to do two clicks. Screenshot from 2024-05-08 15-08-57

Big thanks @leafty for reviewing the code! 🙌 I'll definitely bring up your point about the two clicks needed to remove or update the data source with Jimena. Maybe she can take it into account for the new redesign of Renku 2.0. For now, I think it's crucial to maintain consistency in how we interact with the session launcher and data sources in the side panel. Having uniform styles for those actions.

File: client/src/features/projectsV2/api/storages.api-config.ts
Comment (leafty) [reply client/src/features/projectsV2/api/storages.api-config.ts:1]: This comment still stand:
1. The RTK query tag types need to be updated to `"Storage"` (see "Project", "Namespace" above)
2. The file name and the API name need to be updated t `storage-api` or similar ("The storage API").

File: client/.eslintrc.json (hunk 4 above)
Comment (leafty) [reply client/.eslintrc.json:243]: Can be added.

File: client/src/features/ProjectPageV2/ProjectPageContent/CodeRepositories/RepositoriesBox.tsx [hunk 6]
```diff
@@ -45,7 +45,12 @@ export function CodeRepositoriesDisplay({ project }: { project: Project }) {
       </p>
       <div className={cx("p-2", "ps-3", "pb-0")}>
         {project.repositories?.map((repositoryUrl, index) => (
-          <RepositoryItem key={index} project={project} url={repositoryUrl} />
+          <RepositoryItem
+            key={index}
+            project={project}
+            url={repositoryUrl}
+            showMenu={true}
```
Comment (leafty) [inline client/src/features/ProjectPageV2/ProjectPageContent/CodeRepositories/RepositoriesBox.tsx:52]: nitpick

```suggestion
            showMenu
```

**Review by leafty — APPROVED** [review]
LGTM 👏

**Pull request merged**